components/result-count/src/__snapshots__/test.js.snap

```diff
@@ -1,23 +1,10 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`ResultCount matches snapshot 1`] = `
-.emotion-0 {
-  background: yellow;
-  color: blue;
-  display: inline-block;
-  font-family: Roboto,Arial,sans-serif;
-  font-size: 15px;
-  line-height: 1.25;
-  min-height: 20px;
-  min-width: 20px;
-  padding: 1px 4px;
-  text-align: center;
-}
-
-<span
-  className="emotion-0 emotion-1"
+<StyledResultCount
```

Comment:
I don't think it's necessary to have styles in a snapshot personally since it requires using `mount` - I think this is kind of unit testing _Emotion_ - but rather we should explicitly unit test our components where conditionals are involved by using `jest-emotion` - that gives us 100% coverage.

Reply:
I appreciate though it's actually not only testing Emotion's functionality but also whether defined styles have changed. Hmm.. it's a question of whether it's worth using `mount` across the board for this since we're using so many SC's! Or just `jest-emotion` for explicit tests. Will do a bit of reading about this and try to see what general opinion/approaches are.

Reply:
[This writer](https://medium.com/styled-components/effective-testing-for-styled-components-67982eb7d42b) favours storing the style rules in the snapshot - you well may be right that it's worth it, and we could switch to `mount`, but we'd have to go and refactor a fair few tests. If we have time we could do this soon, if not revisit it later. :) . Thanks for the review, btw!

Reply:
No worries - I completely appreciate the performance point but I believe these points are often raised without considering something like generated component styles - the value of having the styles in the snapshot are quite high and imho are probably worth using the mount. It could be interesting performance testing to see the cost but I’m guessing our component libraries will never be that large that the performance is an issue. Interesting!

Reply:
Yes, even if this PR goes through as-is, I will at some point benchmark use of `mount` and if it's insignificant, as it probably is, I'll update all tests. I agree there's value to be had (they do mention it [here](https://emotion.sh/docs/testing), using `react-test-renderer`). In addition, of course, we can keep the `toHaveStyleRule` stuff which helps us hit our test target 💯!